sqlglot/optimizer/remove_unused_columns.py

```diff
@@ -0,0 +1,162 @@
+from collections import defaultdict
+from copy import copy
+
+from sqlglot import expressions as exp
+from sqlglot.errors import OptimizeError
+from sqlglot.optimizer.helper import select_scope
+
+# Sentinel value that means an outer query selecting ALL columns
+SELECT_ALL = "*"
```

Comment:
let's put this in constants.py

Reply:
I'm a little worried that it doesn't have much meaning outside this module

Reply:
hm ok, up to you